### PR TITLE
fix: add hashes to requirements.txt

### PR DESCRIPTION
**Pull request by SurferJeffAtGoogle**
and update Docker images so they require hashes.

### CODE REVIEW

**Comment by blakeli0** [comment]
Do you plan to fix "pip install" in other places as well? Like [this one](https://github.com/googleapis/synthtool/blob/d229a1258999f599a90a9b674a1c5541e00db588/synthtool/gcp/templates/java_library/.kokoro/release/stage.sh#L19) used for releasing for Java?

**Comment by SurferJeffAtGoogle** [comment]
> Do you plan to fix "pip install" in other places as well? Like [this one](https://github.com/googleapis/synthtool/blob/d229a1258999f599a90a9b674a1c5541e00db588/synthtool/gcp/templates/java_library/.kokoro/release/stage.sh#L19) used for releasing for Java?

Yes.

File: requirements.txt [hunk 1]
```diff
@@ -1,18 +1,309 @@
-setuptools==65.2.0
+#
+# This file is autogenerated by pip-compile with python 3.10
+# To update, run:
+#
+#    pip-compile --allow-unsafe --generate-hashes requirements.in
```
Comment (bcoe) [inline requirements.txt:5]: Is this helpful message automatically inserted by `pip-compile --allow-unsafe --generate-hashes requirements.in`?

**Review by bcoe — APPROVED** [review]

File: requirements.txt (hunk 1 above)
Comment (SurferJeffAtGoogle) [reply requirements.txt:5]: Yes.

**Review by tmatsuo — APPROVED** [review]

**Review by suztomo — APPROVED** [review]
Thank you.

**Comment by suztomo** [comment]
Oh, change for java is gone? https://github.com/googleapis/synthtool/commit/0a5fa6744eacc0968356cc3100943303a853e7e8

**Comment by SurferJeffAtGoogle** [comment]
> Oh, change for java is gone? [0a5fa67](https://github.com/googleapis/synthtool/commit/0a5fa6744eacc0968356cc3100943303a853e7e8)

Yeah, sorry.  Java requires some more work upstream.

**Pull request merged**